modules/apps/portal-search-solr8/portal-search-solr8-impl/src/main/java/com/liferay/portal/search/solr8/internal/connection/SolrClientManager.java

```diff
@@ -91,26 +91,6 @@ protected synchronized void deactivate(Map<String, Object> properties) {
 		_close();
 	}
 
-	@Reference(
-		policy = ReferencePolicy.DYNAMIC,
-		policyOption = ReferencePolicyOption.GREEDY, target = "(type=BASIC)"
-	)
-	protected void setBasicHttpClientFactory(
```

Comment:
@slnn Is it really safe to do this? The default cardinality is MANDATORY, which is 1..1 relationship, which means this component will not start, until this reference is satisfied.

So the logic is, it depends on BASIC and CERT `HttpClientFactory`, and also supports other types.